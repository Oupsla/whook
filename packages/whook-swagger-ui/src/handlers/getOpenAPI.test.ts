import initGetOpenAPI from './getOpenAPI';

describe('getOpenAPI', () => {
  const API = {
    swagger: '2.0',
    paths: {
      '/time': {
        options: {
          tags: ['public'],
          'x-whook': { memx: 2, tx: 18 },
        },
        get: {
          tags: ['public'],
          'x-whook': { memx: 2, tx: 18 },
        },
        put: {
          tags: ['private'],
          'x-whook': { private: true },
        },
      },
    },
    tags: [{ name: 'public' }, { name: 'private' }],
  };

  it('should work', async () => {
    const getOpenAPI = await initGetOpenAPI({
      API,
    });
    const response = await getOpenAPI({});

    expect({
      response: {
        ...response,
        body: {
          ...response.body,
          info: {
            ...response.body.info,
            version: '<already_tested>',
          },
        },
      },
    }).toMatchSnapshot();
  });

  it('should show every endpoints when authenticated', async () => {
    const getOpenAPI = await initGetOpenAPI({
      API,
    });
    const response = await getOpenAPI({
      authenticated: true,
    });

    expect({
      response: {
        ...response,
        body: {
          ...response.body,
          info: {
            ...response.body.info,
            version: '<already_tested>',
          },
        },
      },
    }).toMatchSnapshot();
  });
});
