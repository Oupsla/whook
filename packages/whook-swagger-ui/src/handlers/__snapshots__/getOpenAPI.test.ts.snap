// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getOpenAPI should show every endpoints when authenticated 1`] = `
Object {
  "response": Object {
    "body": Object {
      "info": Object {
        "version": "<already_tested>",
      },
      "paths": Object {
        "/time": Object {
          "get": Object {
            "tags": Array [
              "public",
            ],
            "x-whook": Object {
              "memx": 2,
              "tx": 18,
            },
          },
          "put": Object {
            "tags": Array [
              "private",
            ],
            "x-whook": Object {
              "private": true,
            },
          },
        },
      },
      "swagger": "2.0",
      "tags": Array [
        Object {
          "name": "public",
        },
        Object {
          "name": "private",
        },
      ],
    },
    "status": 200,
  },
}
`;

exports[`getOpenAPI should work 1`] = `
Object {
  "response": Object {
    "body": Object {
      "info": Object {
        "version": "<already_tested>",
      },
      "paths": Object {
        "/time": Object {
          "get": Object {
            "tags": Array [
              "public",
            ],
            "x-whook": undefined,
          },
        },
      },
      "swagger": "2.0",
      "tags": Array [
        Object {
          "name": "public",
        },
        Object {
          "name": "private",
        },
      ],
    },
    "status": 200,
  },
}
`;
